vassal-app/src/main/java/VASSAL/counters/Decorator.java

```diff
@@ -136,6 +193,26 @@ else if (Properties.SELECTED.equals(key)) {
     }
   }
 
+  /**
+   * Within a Trait/Decorator, the default behavior when setting a property is to handle
+   * changing our own inner/outer links directly, to cache the selection state (while also
+   * passing it inward), and to simply pass every other property change request inward. A
+   * Trait with its own properties to maintain would intercept and process those requests,
+   * while sending the rest inward to the next member of the piece.
+   *
+   * <br><br>Properties can be associated with a piece -- many may be game-specific, but others
+   * are standard, such as the LocationName property exposed by BasicPiece -- and can
+   * be set through this interface. The properties may or may not need to be encoded in
+   * the piece's {@link #getState} method. Properties include the value of e.g. {@link Marker}
+   * Traits, {@link DynamicProperty} Traits, and so forth.
+   *
+   * <br><br><b>NOTE:</b> Properties outside the piece, however, CANNOT be set by this
+   * method (e.g. Global Properties), even though they can be read by {@link #getProperty} --
+   * in this the two methods are not perfect mirrors.
+   *
```

Comment:
Possibly note also that many properties maintained by traits cannot be set by calling SetProperty, you need to check the code of the individual traits to see what they support.